src/resources/parser/glb-parser.js

```diff
@@ -887,9 +890,18 @@ const createMesh = function (device, gltfMesh, accessors, bufferViews, callback,
                 mesh.primitive[0].count = vertexBuffer.numVertices;
             }
 
-            // TODO: Refactor, we should not store temporary data on the mesh.
-            // The container should store some mapping table instead.
-            mesh.materialIndex = primitive.material;
+            if (primitive.hasOwnProperty("extensions") && primitive.extensions.hasOwnProperty("KHR_materials_variants")) {
+                const variants = primitive.extensions.KHR_materials_variants;
+                const tempMapping = {};
+                variants.mappings.forEach((mapping) => {
+                    mapping.variants.forEach((variant) => {
+                        tempMapping[variant] = mapping.material;
+                    });
+                });
+                meshVariants[mesh.id] = tempMapping;
```

Comment:
I'd suggest meshVariants and meshDefaultMaterials to use object (with a key lookup), or a Map, instead of being an array. If the user has lots of meshes, the mesh.id will be high and allocate large arrays for no reason.

Reply:
Does it become an array? I am assigning a member to a value, doesn't that just add a member to an object?

Reply:
you initialise them to be arrays in the code below I think?
```
    const meshVariants = [];
    const meshDefaultMaterials = [];
```
you could change those to be objects { }, but I think those are being passed to some functions that expect arrays and look them up by index, so it would not work I think.